compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -5856,31 +5856,92 @@ private void handleSafeNavigation(BLangAccessExpression accessExpr, BType type,
 
         BLangMatch matchStmt = ASTBuilderUtil.createMatchStatement(accessExpr.pos, accessExpr.expr, new ArrayList<>());
 
+        boolean allRecords = false;
+        BUnionType errorLiftedType = BUnionType.create(null, new LinkedHashSet<>());
```

Comment:
Can we work with a `LinkedHashSet` instead? We don't need the union type right?

Reply:
Done